### PR TITLE
PIA-491: Display the digital invoice if the line items are present.

### PR DESCRIPTION
**Pull request by maciejtrybilo**

### CODE REVIEW

File: GiniVision/Classes/Networking/GiniNetworkingScreenAPICoordinator.swift [hunk 1]
```diff
@@ -136,23 +135,44 @@ final class GiniNetworkingScreenAPICoordinator: GiniScreenAPICoordinator {
...
-            case .success(let extractions):
-                self.deliver(result: extractions, analysisDelegate: networkDelegate)
+            case .success(let extractionResult):
+                
+                DispatchQueue.main.async {
+                    do {
+                        try self.showDigitalInvoiceScreen(extractionResult: extractionResult, analysisDelegate: networkDelegate)
+                    } catch {
+                        self.deliver(result: extractionResult, analysisDelegate: networkDelegate)
+                    }
```
Comment (a-szotyori) [inline GiniVision/Classes/Networking/GiniNetworkingScreenAPICoordinator.swift]: Could you make the decision to show the digital invoice screen more explicit? I did it by checking whether the `lineItems` compound extraction label is present or not.

You can still keep the `do - catch` of course.

File: Example/Example Swift/ComponentAPICoordinator.swift [hunk 2]
```diff
@@ -679,12 +681,22 @@ extension ComponentAPICoordinator {
...
+            let images = self.pages.compactMap { $0.document.previewImage }
+            let extractions: [String: Extraction] = Dictionary(uniqueKeysWithValues: extractionResult.extractions.compactMap {
+                guard let name = $0.name else { return nil }
+                
+                return (name, $0)
+            })
+            
+            let result = AnalysisResult(extractions: extractions, lineItems: extractionResult.lineItems, images: images)
+            
+            showResultsTableScreen(with: result)
```
Comment (a-szotyori) [inline Example/Example Swift/ComponentAPICoordinator.swift]: You could also check here for `extractionResult.lineItems` and show the digital invoice screen, if it's present, no?

File: GiniVision/Classes/Core/Screens/Digital Invoice/DigitalInvoice.swift [hunk 3]
```diff
@@ -98,3 +72,38 @@ extension DigitalInvoice {
...
+    enum DigitalInvoiceParsingException: Error {
+        case lineItemsMissing
+        case nameMissing
+        case quantityMissing
+        case priceMissing
+        case articleNumberMissing
+        case mixedCurrenciesInOneInvoice
+        case cannotParseQuantity(string: String)
+        case cannotParsePrice(string: String)
+    }
```
[the remark this answers is not in the document]
Comment (a-szotyori) [reply GiniVision/Classes/Core/Screens/Digital Invoice/DigitalInvoice.swift]: Found where they are caught 😁.

[the remark this answers is not in the document]
Comment (maciejtrybilo) [reply GiniVision/Classes/Core/Screens/Digital Invoice/DigitalInvoice.swift]: The thing with falling back to default values is that those are meaningless for the user. The parsing is strict so we can bail out on showing the return assistant if the data is wrong for some reason.

[the remark this answers is not in the document]
Comment (a-szotyori) [reply GiniVision/Classes/Core/Screens/Digital Invoice/DigitalInvoice.swift]: Makes perfect sense. So we only show the return assistant if these conditions are fulfilled. I will apply this to the Android version, too.

File: GiniVision/Classes/Core/Screens/Digital Invoice/LineItem.swift [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            
+            enum Reason: String, CaseIterable {
+                case looksDifferent
+                case poorQualityOrFaulty
+                case doesNotFit
+                case doesNotSuit
+                case wrongItem
+                case damaged
+                case arrivedTooLate
+            }
```
[the remark this answers is not in the document]
Comment (maciejtrybilo) [reply GiniVision/Classes/Core/Screens/Digital Invoice/LineItem.swift]: I've taken it a bit closer to a possible dynamic solution.

[the remark this answers is not in the document]
Comment (a-szotyori) [reply GiniVision/Classes/Core/Screens/Digital Invoice/LineItem.swift]: 👍

File: GiniVision/Classes/Networking/GiniNetworkingScreenAPICoordinator.swift (hunk 1 above)
Comment (maciejtrybilo) [reply GiniVision/Classes/Networking/GiniNetworkingScreenAPICoordinator.swift]: Updated. Does showing DigitalInvoice initialisation show better what's going on?

File: Example/Example Swift/ComponentAPICoordinator.swift (hunk 2 above)
Comment (maciejtrybilo) [reply Example/Example Swift/ComponentAPICoordinator.swift]: Very well spotted and a few things fixed for Component API thanks to this 👍🏼

File: GiniVision/Classes/Networking/GiniNetworkingScreenAPICoordinator.swift (hunk 1 above)
Comment (a-szotyori) [reply GiniVision/Classes/Networking/GiniNetworkingScreenAPICoordinator.swift]: Thank you! Looks great!

File: Example/Example Swift/ComponentAPICoordinator.swift (hunk 2 above)
Comment (a-szotyori) [reply Example/Example Swift/ComponentAPICoordinator.swift]: Nice! 👍

**Pull request merged**